uat/custom-components/client-ipc/src/main/java/com/aws/greengrass/testing/ipc/LocalIpcSubscriber.java

```diff
@@ -0,0 +1,153 @@
+import software.amazon.awssdk.aws.greengrass.model.SubscriptionResponseMessage;
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import javax.inject.Inject;
+
+@Log4j2
+public class LocalIpcSubscriber implements Consumer<String[]> {
+
+    private static final int DEFAULT_PORT = 8033;
+    private static final String TOPICS_SEPARATOR = ",";
+
+    private static final String DEFAULT_CONTEXT = "ReceivedPubsubMessage";
+
+    private String assertionServerUrl = "http://localhost:8080";
+
+    private final ExecutorService executor = Executors.newCachedThreadPool();
+
+    @Inject
+    public LocalIpcSubscriber() {
+    }
+
+    /**
+     * Start LocalIpcSubscriber execution.
+     *
+     * @param args command line args
+     */
+    @Override
+    public void accept(String[] args) {
+        log.info("Args: {}", args);
+        List<String> topics = Arrays.asList(args[0].split(TOPICS_SEPARATOR));
+        log.info("Subscribe to topics: {}", topics);
+        topics.forEach(topic -> executor.execute(() -> subscribe(topic)));
+        final String url = Optional.ofNullable(args[1])
+                                   .orElse(assertionServerUrl);
+        log.info("Assertion server URL: {}", url);
+        assertionServerUrl = url;
+    }
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private void subscribe(String topic) {
+        try (GreengrassCoreIPCClientV2 ipcClient = GreengrassCoreIPCClientV2.builder()
+                                                                            .withPort(DEFAULT_PORT)
+                                                                            .build()) {
+            SubscribeToTopicRequest request = new SubscribeToTopicRequest().withTopic(topic);
+            GreengrassCoreIPCClientV2.StreamingResponse<SubscribeToTopicResponse, SubscribeToTopicResponseHandler>
+                    response = ipcClient.subscribeToTopic(request,
+                                                          this::onStreamEvent,
+                                                          Optional.of(this::onStreamError),
+                                                          Optional.of(this::onStreamClosed));
+            SubscribeToTopicResponseHandler responseHandler = response.getHandler();
+            log.info("Successfully subscribed to topic: {}", topic);
+
+            try {
+                while (true) {
+                    Thread.sleep(10_000);
+                }
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

**Remember**: Do this only for InterruptedException. Doing this for other types of exceptions may give rise to subtle bugs.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)